### PR TITLE
Multiplotting and update subplots

### LINKED ISSUE #24674: multi plotting update locations
Adding in the auto arrangement of the subplots into multiplotting

### PR DESCRIPTION
**Pull request by AnthonyLim23**
This adds the ability to auto-update the layout of the multi plot widget. 

**Report to:** [user name]/[nobody]. 

**To test:**
Enable the plotting_test gui
Open the plotting_test gui
You can use the button at the bottom to add subplots
You can remove lines from the subplots (when the number of lines reaches zero the whole plot is removed)
You can remove a whole subplot from the toolbar
If the GUI has zero subplots it will close. 

Fixes #24674. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/MultiPlotting/navigation_toolbar.py [hunk 1]
```diff
@@ -15,9 +15,13 @@ def __init__(self, *args, **kwargs):
         ic = QtGui.QIcon(pm)
         # self.add = self.addAction(ic, "Add line")
```
Comment (Mantid-Matthew) [inline scripts/MultiPlotting/navigation_toolbar.py]: I assume the plan is to add the Add line action later?

**Comment by Mantid-Matthew** [comment]
I can add and remove subplots and the grid resizes appropriately. Currently I cannot get the GUI to open in a python3 build however.

File: scripts/MultiPlotting/gridspec_engine.py [hunk 2]
```diff
@@ -29,9 +29,10 @@ def __init__(self, grid=defaultGridspecGrid, max_plot=None):
     def getGridSpec(self, number):
         if number <= 0:
             return
-        if number > self._max_plot and self._max_plot is not None:
```
Comment (Mantid-Matthew) [inline scripts/MultiPlotting/gridspec_engine.py]: I think just switching the order of these two checks should have the same effect as nesting them.

**Comment by Mantid-Matthew** [comment]
Currently I can't get autoscale Y to work when I have selected just one subplot it always seems to scale all the plots at the same time.

**Comment by AnthonyLim23** [comment]
the auto update only touches the selected plot for me

**Comment by AnthonyLim23** [comment]
@mantidproject/gatekeepers could we get this merged please? The build failure is unrelated

**Comment by AnthonyLim23** [comment]
see #24879 for autoscale 'bug'

**Review by Matthew-Andrew — APPROVED** [review]
Happy for the re-scaling issue to be dealt with later

**Pull request merged**